lib/topological_inventory/ansible_tower/collectors_pool.rb

```diff
@@ -18,6 +18,16 @@ def path_to_secrets
       File.expand_path("../../../secret", File.dirname(__FILE__))
     end
 
+    def source_valid?(source, secret)
```

Comment:
this should be extended to actually validate the credentials, correct?

Reply:
it validates if credentials are present, they are in `secret`. Or do you mean validations that credentials are valid? I think this is similar to Source availability checker, which implements @abellotti - possibly if it'll be in db, then that source won't be event deployed (after merge of that)